Controllers/PostController.cs

```diff
@@ -198,8 +198,86 @@ public IActionResult Create(PostCreateDTO post)
                 }
             })
             .SingleOrDefault(p => p.Id == newPost.Id);
-        
+
         return Created($"posts/{createdPost.Id}", createdPost);
     }
 
+    [HttpGet("unapproved")]
+    public IActionResult GetUnapprovedAndPublished()
+    {
+
+        List<Post> postList = _dbContext.Posts
+             .Where(p => p.IsApproved == false && (p.PublicationDate.Value <= DateTime.Now | p.PublicationDate == null))
```

Comment:
Include all unapproved, not only those that would be published already if they were approved